File: Examples/Outlook/C#/NetOffice COMAddin Examples/Extended/SamplePane.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace NetOfficeTools.ExtendedOutlookCS4
{
    public partial class SamplePane : UserControl, NetOffice.OutlookApi.Tools.ITaskPane
    {
        #region Ctor

        public SamplePane()
        {
            InitializeComponent();
        }

        #endregion

        #region Properties

        private DateTime StartTime { get; set; }

        #endregion

        #region ITaskpane

        public void OnConnection(NetOffice.OutlookApi.Application application, NetOffice.OfficeApi._CustomTaskPane parentPane, object[] customArguments)
        {
            StartTime = DateTime.Now;
            buttonEnabled_Click(buttonEnabled, new EventArgs());
        }

        public void OnDisconnection()
        {

        }

        #endregion

        #region UI Trigger

        private void buttonEnabled_Click(object sender, EventArgs e)
        {
            if (timerRunningTime.Enabled)
            {
                timerRunningTime.Enabled = false;
                buttonEnabled.Text = "Enable";
                buttonEnabled.ImageKey = "alarmclock_run.png";
                labelTime.ForeColor = Color.FromKnownColor(KnownColor.ControlText);
            }
            else
            {
                timerRunningTime.Enabled = true;
                buttonEnabled.Text = "Disable";
                buttonEnabled.ImageKey = "alarmclock_stop.png";
                labelTime.ForeColor = Color.White;
            }
        }

        private void buttonReset_Click(object sender, EventArgs e)
        {
            StartTime = DateTime.Now;
            labelTime.Text = "00:00:00";
        }

        private void timerRunningTime_Tick(object sender, EventArgs e)
        {
            TimeSpan ts = DateTime.Now - StartTime;
            labelTime.Text = String.Format("{0:00}:{1:00}:{2:00}", ts.Hours, ts.Minutes, ts.Seconds);

        }

        #endregion
    }
}

File: Examples/Outlook/VB/NetOffice COMAddin Examples/Extended/SamplePane.vb
Imports NetOffice.OutlookApi.Tools

Public Class SamplePane
    Implements ITaskPane

#Region "Properties"

    Private StartTime As DateTime

#End Region

#Region "ITaskPane"

    Public Sub OnConnection(ByVal application As NetOffice.OutlookApi.Application, ByVal parentPane As NetOffice.OfficeApi._CustomTaskPane, ByVal customArguments() As Object) Implements NetOffice.OutlookApi.Tools.ITaskPane.OnConnection

        StartTime = DateTime.Now
        buttonEnabled_Click(buttonEnabled, New EventArgs())

    End Sub

    Public Sub OnDisconnection() Implements NetOffice.OutlookApi.Tools.ITaskPane.OnDisconnection

    End Sub

#End Region

#Region "Trigger"

    Private Sub buttonEnabled_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonEnabled.Click

        If (timerRunningTime.Enabled) Then
            timerRunningTime.Enabled = False
            buttonEnabled.Text = "Enable"
            buttonEnabled.ImageKey = "alarmclock_run.png"
            labelTime.ForeColor = System.Drawing.Color.FromKnownColor(System.Drawing.KnownColor.ControlText)

        Else
            timerRunningTime.Enabled = True
            buttonEnabled.Text = "Disable"
            buttonEnabled.ImageKey = "alarmclock_stop.png"
            labelTime.ForeColor = System.Drawing.Color.White
        End If

    End Sub

    Private Sub buttonReset_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonReset.Click

        StartTime = DateTime.Now
        labelTime.Text = "00:00:00"

    End Sub

    Private Sub timerRunningTime_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles timerRunningTime.Tick

        Dim ts As TimeSpan = DateTime.Now - StartTime
        labelTime.Text = String.Format("{0:00}:{1:00}:{2:00}", ts.Hours, ts.Minutes, ts.Seconds)

    End Sub

#End Region

End Class
